### PR TITLE
Dialog: separate into standalone modules

### PR DESCRIPTION
**Pull request by ianmcburnie**
DO NOT SQUASH.

Issue #1183 
Issue #1175 
Issue #1176 

Splits dialog into following modules:

* Fullscreen-Dialog
* Lightbox-Dialog (includes mini, alert & confirm variants)
* Panel-Dialog

These two types of dialogs were already separate:

* Drawer
* Toast

Haven't fully decided on whether all named need "-dialog" postfix or not. We can decide later.

Note, I wont be doing any major cleanup/refactoring of the selectors and styles in this PR (such as changing out absolute positioning for flexbox) - we can do that in a follow up.

Also, I've not tackled updating the stories yet.

Screen Shot 2020-07-20 at 1 46 06 PM
Screen Shot 2020-07-20 at 1 46 15 PM
Screen Shot 2020-07-20 at 1 46 21 PM
Screen Shot 2020-07-20 at 1 46 26 PM
Screen Shot 2020-07-20 at 1 46 34 PM

### CODE REVIEW

File: src/less/lightbox-dialog/base/lightbox-dialog.less [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    .lightbox-dialog__window--fade {
+        opacity: 1;
+    }
+}
+
+// MEDIA QUERIES
+//-----------------------------
+@media (max-width: 600px) {
+    .lightbox-dialog__window {
+        .create-grid(@cols: 8, @gutter: 8px, @gutter-exponent: 4, @size: 8, @min: 8, @max: 8);
```
Comment (agliga) [inline src/less/lightbox-dialog/base/lightbox-dialog.less:121]: Is there a way we can remove this or change it? Would like to see no more custom grids around

File: src/less/lightbox-dialog/base/lightbox-dialog.less [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+.lightbox-dialog[role="dialog"] {
+    .dialog-base();
+}
+
+.lightbox-dialog__window {
+    .dialog-window();
+
+    flex-direction: column;
+    left: 0;
+    max-width: 616px;
```
Comment (agliga) [inline src/less/lightbox-dialog/base/lightbox-dialog.less:12]: Any reason this was moved out of mixins. This is common across all dialogs except for mini
Maybe change it to a variable?

File: src/less/lightbox-dialog/base/lightbox-dialog.less (hunk 1 above)
Comment (ianmcburnie) [reply src/less/lightbox-dialog/base/lightbox-dialog.less:121]: I would like to remove this too, but I'm not completely familiar with its purpose. I think next PR will look into things like this.

File: src/less/lightbox-dialog/base/lightbox-dialog.less (hunk 2 above)
Comment (ianmcburnie) [reply src/less/lightbox-dialog/base/lightbox-dialog.less:12]: Stylelint was complaining of duplicate max-width in the CSS. I think the mixins can be cleaned up and refactored in the next PR.

**Comment by agliga** [comment]
Looks good to merge

**Pull request merged**